test/common/dashboards_exist.go

```diff
@@ -0,0 +1,125 @@
+package common
+
+import (
+	goctx "context"
+	"encoding/json"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type dashboardsTestRule struct {
+	Title string `json:"title"`
+}
+
+var expectedDashboards = []dashboardsTestRule{
+	{
+		Title: "Endpoints Detailed",
+	},
+	{
+		Title: "Endpoints Report",
+	},
+	{
+		Title: "Endpoints Summary",
+	},
+	{
+		Title: "Keycloak",
+	},
+	{
+		Title: "Resource Usage By Namespace",
+	},
+	{
+		Title: "Resource Usage By Pod",
+	},
+	{
+		Title: "Resource Usage for Cluster",
+	},
+	{
+		Title: "Syndesis - Infra - API",
+	},
+	{
+		Title: "Syndesis - Infra - DB",
+	},
+	{
+		Title: "Syndesis - Infra - Home",
+	},
+	{
+		Title: "Syndesis - Infra - JVM",
+	},
+	{
+		Title: "Syndesis - Integrations - Camel",
+	},
+	{
+		Title: "Syndesis - Integrations - Home",
+	},
+	{
+		Title: "Syndesis - Integrations - JVM",
+	},
+	{
+		Title: "UnifiedPush Operator",
+	},
+	{
+		Title: "UnifiedPush Server",
+	},
+}
+
+func TestIntegreatlyDashboardsExist(t *testing.T, ctx *TestingContext) {
+	pods := &corev1.PodList{}
+	opts := []k8sclient.ListOption{
+		k8sclient.InNamespace(namespacePrefix + "middleware-monitoring-operator"),
+		k8sclient.MatchingLabels{"app": "grafana"},
+	}
+
+	err := ctx.Client.List(goctx.TODO(), pods, opts...)
+	if err != nil {
+		t.Fatal("failed to list pods", err)
+	}
+
+	if len(pods.Items) != 1 {
+		t.Fatal("grafana pod not found")
+	}
+
+	output, err := execToPod("curl localhost:3000/api/search",
+		pods.Items[0].ObjectMeta.Name,
+		namespacePrefix+"middleware-monitoring-operator",
+		"grafana", ctx)
+	if err != nil {
+		t.Fatal("failed to exec to pod:", err)
+	}
+
+	var grafanaApiCallOutput []dashboardsTestRule
+	err = json.Unmarshal([]byte(output), &grafanaApiCallOutput)
+	if err != nil {
+		t.Logf("failed to unmarshall json: %s", err)
+	}
+
+	if len(grafanaApiCallOutput) == 0 {
+		t.Fatal("no grafana dashboards were found : %w", grafanaApiCallOutput)
+	}
+
+	var expectedDashboardTitles []string
+	for _, dashboard := range expectedDashboards {
+		expectedDashboardTitles = append(expectedDashboardTitles, dashboard.Title)
+	}
+	var actualDashboardTitles []string
+	for _, dashboard := range grafanaApiCallOutput {
+		actualDashboardTitles = append(actualDashboardTitles, dashboard.Title)
+	}
+
+	dashboardDiffUnexpected := difference(actualDashboardTitles, expectedDashboardTitles)
+	dashboardDiffMissing := difference(expectedDashboardTitles, actualDashboardTitles)
+
+	if len(dashboardDiffUnexpected) > 0 {
+		t.Logf("unexpected dashboards found: %s", strings.Join(dashboardDiffUnexpected, ", "))
```

Comment:
Tip: using Errorf the test will be marked as failed but the execution will proceed so that it will also log the missing dashboards and there is no need for the last Fatal
```suggestion
		t.Errorf("unexpected dashboards found: %s", strings.Join(dashboardDiffUnexpected, ", "))
```

Reply:
Even better you could loop over the unexpected dashboard and Errorf each one